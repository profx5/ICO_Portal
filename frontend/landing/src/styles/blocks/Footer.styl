.Footer
    background #FAFAFA
    &_container
        padding-top 70px
        padding-bottom 45px
        @media xs
            padding-top 40px
            padding-bottom 55px
    &_top
        @media smPlus
            display flex
            flex-flow row nowrap
            justify-content space-between
            margin-bottom 23px
        @media xs
            order 2
            position absolute
            width 100%
            left 50%
            transform translateX(-50%)
            top 68px
    &_menuItem
        color #383838
        display inline-block
        font-weight 400
        margin-right 20px
        transition color .25s ease
        &:hover
            color pink
    &_link
        font-weight 400
        @media xs
            display block
            text-align center
    &_divider
        background #959595
        opacity .2
        height 1px
        margin-bottom 23px
        @media xs
            display none
    &_info
        display: flex
        flex-flow row nowrap
        justify-content space-between
        padding-bottom 10px
        @media xs
            flex-wrap wrap
            order 1
    &_addressText
        font-weight 400
        font-style normal
        &:before
            content url('../img/common/svg/location.svg')
            margin-right 7px
            @media xs
                display none
        @media xs
            text-align center
    &_socialWrapper
        display flex
        flex-flow row nowrap
        justify-content space-between
        width 145px
    &_address
        @media xs
            order 3
            border-top 1px solid rgba(149,149,149,.2)
            padding-top 25px
    &_links
        @media xs
            margin 0 auto
            order 1
            margin-bottom 50px
    &_copyright
        display: flex
        flex-flow row nowrap
        justify-content space-between
        @media xs
            flex-wrap wrap
            justify-content center
    &_rightText
        position relative
        padding-left 30px
        white-space pre
        &:before
            content ''
            height 67%
            width 1px
            background black
            position absolute
            left 15px
            top 50%
            transform translateY(-50%)
    &_copyrightText
        font-weight 400
        @media xs
            order 2
    &_copyrightBlock
        font-weight 400
        display: flex
        flex-flow row nowrap
        @media xs
            order 1
    &_iconWrapper svg path
        transition fill .2s ease
    &_iconWrapper:hover svg path
        fill blue