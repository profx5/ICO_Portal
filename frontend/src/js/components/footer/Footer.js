import React from 'react';


export default class Footer extends React.Component {
    render () {
        return (
            <footer className="Footer col-md-12"></footer>
        );
    }
}