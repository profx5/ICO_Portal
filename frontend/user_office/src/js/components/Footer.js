import React from 'react';

const Footer = () => (
    <footer className="Footer col-md-12"></footer>
)

export default Footer
