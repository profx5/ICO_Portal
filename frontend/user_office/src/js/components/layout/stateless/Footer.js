import React from 'react';
import styled from 'styled-components';
import {media} from 'js/services/media';

import {Link} from 'react-router-dom';

const Footer = ({}) => {
    return (
        <Wrapper>
            <ul>
                <li><Link to="/user_office">White Paper</Link></li>
                <li><Link to="/user_office">ICO site</Link></li>
                <li><Link to="/user_office">User Agreement</Link></li>
            </ul>
            <Copyright>Â© 2018 Ongrid.pro</Copyright>
        </Wrapper>
    )
}


export default Footer;

const Wrapper = styled.div`
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: flex-end;
    flex-basis: 100%;
    text-align: center;
    margin-bottom: 40px;
    margin-top: 40px;
    ul {
        margin-bottom: 14px;
        li {
            margin-bottom: 5px;
            display: inline-block;
            ${media.xs} {
                display: block;
            }
            &:nth-of-type(2) {
                ${media.smPlus} {
                    margin: 0 25px;
                }
            }
            a {
                font-size: 14px;
                color: rgba(0,0,0,.5);
                transition: color .25s ease;
                ${media.xs} {
                    color: rgb(0,0,0);
                }
                &:hover {
                    color: rgb(0,0,0);
                }
            }
        }
    }
`;

const Copyright = styled.div`
    font-size: 14px;
    ${media.xs} {
        color: rgb(53, 119, 252);
    }
`;
