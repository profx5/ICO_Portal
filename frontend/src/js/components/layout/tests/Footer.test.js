import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Footer from 'js/components/layout/stateless/Footer';


configure({ adapter: new Adapter() });

it('should render stateless Footer', () => {
    const component = shallow(<Footer/>);
    expect(component).toMatchSnapshot();
})
