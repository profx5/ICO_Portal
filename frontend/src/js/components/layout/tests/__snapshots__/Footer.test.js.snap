// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless Footer 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ul>
          <li>
            <Link
              replace={false}
              to="/user_office"
            >
              White Paper
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/user_office"
            >
              ICO site
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/user_office"
            >
              User Agreement
            </Link>
          </li>
        </ul>,
        <ForwardRef>
          © 2018 Ongrid.pro
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                replace={false}
                to="/user_office"
              >
                White Paper
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/user_office"
              >
                ICO site
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/user_office"
              >
                User Agreement
              </Link>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/user_office"
              >
                White Paper
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "White Paper",
                "replace": false,
                "to": "/user_office",
              },
              "ref": null,
              "rendered": "White Paper",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/user_office"
              >
                ICO site
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "ICO site",
                "replace": false,
                "to": "/user_office",
              },
              "ref": null,
              "rendered": "ICO site",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/user_office"
              >
                User Agreement
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "User Agreement",
                "replace": false,
                "to": "/user_office",
              },
              "ref": null,
              "rendered": "User Agreement",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "© 2018 Ongrid.pro",
        },
        "ref": null,
        "rendered": "© 2018 Ongrid.pro",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
    font-size: 14px;
    ",
              "@media (max-width: 767px)",
              " {
        color: rgb(53, 119, 252);
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: flex-end;
    flex-basis: 100%;
    text-align: center;
    margin-bottom: 40px;
    margin-top: 40px;
    ul {
        margin-bottom: 14px;
        li {
            margin-bottom: 5px;
            display: inline-block;
            ",
          "@media (max-width: 767px)",
          " {
                display: block;
            }
            &:nth-of-type(2) {
                ",
          "@media (min-width: 768px)",
          " {
                    margin: 0 25px;
                }
            }
            a {
                font-size: 14px;
                color: rgba(0,0,0,.5);
                transition: color .25s ease;
                ",
          "@media (max-width: 767px)",
          " {
                    color: rgb(0,0,0);
                }
                &:hover {
                    color: rgb(0,0,0);
                }
            }
        }
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ul>
            <li>
              <Link
                replace={false}
                to="/user_office"
              >
                White Paper
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/user_office"
              >
                ICO site
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/user_office"
              >
                User Agreement
              </Link>
            </li>
          </ul>,
          <ForwardRef>
            © 2018 Ongrid.pro
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  replace={false}
                  to="/user_office"
                >
                  White Paper
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/user_office"
                >
                  ICO site
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/user_office"
                >
                  User Agreement
                </Link>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/user_office"
                >
                  White Paper
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "White Paper",
                  "replace": false,
                  "to": "/user_office",
                },
                "ref": null,
                "rendered": "White Paper",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/user_office"
                >
                  ICO site
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "ICO site",
                  "replace": false,
                  "to": "/user_office",
                },
                "ref": null,
                "rendered": "ICO site",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/user_office"
                >
                  User Agreement
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "User Agreement",
                  "replace": false,
                  "to": "/user_office",
                },
                "ref": null,
                "rendered": "User Agreement",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "© 2018 Ongrid.pro",
          },
          "ref": null,
          "rendered": "© 2018 Ongrid.pro",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
    font-size: 14px;
    ",
                "@media (max-width: 767px)",
                " {
        color: rgb(53, 119, 252);
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: flex-end;
    flex-basis: 100%;
    text-align: center;
    margin-bottom: 40px;
    margin-top: 40px;
    ul {
        margin-bottom: 14px;
        li {
            margin-bottom: 5px;
            display: inline-block;
            ",
            "@media (max-width: 767px)",
            " {
                display: block;
            }
            &:nth-of-type(2) {
                ",
            "@media (min-width: 768px)",
            " {
                    margin: 0 25px;
                }
            }
            a {
                font-size: 14px;
                color: rgba(0,0,0,.5);
                transition: color .25s ease;
                ",
            "@media (max-width: 767px)",
            " {
                    color: rgb(0,0,0);
                }
                &:hover {
                    color: rgb(0,0,0);
                }
            }
        }
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
